eclair-core/src/main/scala/fr/acinq/eclair/payment/send/PaymentLifecycle.scala

```diff
@@ -266,6 +240,49 @@ class PaymentLifecycle(nodeParams: NodeParams, cfg: SendPaymentConfig, router: A
+    data.route.getChannelUpdateForNode(nodeId) match {
+      case Some(u) if u.shortChannelId != failure.update.shortChannelId =>
+        // it is possible that nodes in the route prefer using a different channel (to the same N+1 node) than the one we requested, that's fine
+        log.info(s"received an update for a different channel than the one we asked: requested=${u.shortChannelId} actual=${failure.update.shortChannelId} update=${failure.update}")
+      case Some(u) if Announcements.areSame(u, failure.update) =>
+        // node returned the exact same update we used, this can happen e.g. if the channel is imbalanced
+        // in that case, let's temporarily exclude the channel from future routes, giving it time to recover
+        log.info(s"received exact same update from nodeId=$nodeId, excluding the channel from futures routes")
+        val nextNodeId = data.route.hops.find(_.nodeId == nodeId).get.nextNodeId
+        router ! ExcludeChannel(ChannelDesc(u.shortChannelId, nodeId, nextNodeId))
+      case Some(u) if PaymentFailure.hasAlreadyFailedOnce(nodeId, data.failures) =>
+        // this node had already given us a new channel update and is still unhappy, it is probably messing with us, let's exclude it
+        log.warning(s"it is the second time nodeId=$nodeId answers with a new update, excluding it: old=$u new=${failure.update}")
+        val nextNodeId = data.route.hops.find(_.nodeId == nodeId).get.nextNodeId
+        router ! ExcludeChannel(ChannelDesc(u.shortChannelId, nodeId, nextNodeId))
+      case Some(u) =>
+        log.info(s"got a new update for shortChannelId=${u.shortChannelId}: old=$u new=${failure.update}")
+      case None =>
+        log.error(s"couldn't find a channel update for node=$nodeId, this should never happen")
+    }
+    // in any case, we forward the update to the router: if the channel is disabled, the router will remove it from its routing table
+    router ! failure.update
+    // we return updated assisted routes: they take precedence over the router's routing table
+    if (Announcements.isEnabled(failure.update.channelFlags)) {
+      data.c.assistedRoutes.map(_.map {
+        case extraHop: ExtraHop if extraHop.shortChannelId == failure.update.shortChannelId => extraHop.copy(
+          cltvExpiryDelta = failure.update.cltvExpiryDelta,
+          feeBase = failure.update.feeBaseMsat,
+          feeProportionalMillionths = failure.update.feeProportionalMillionths
+        )
+        case extraHop => extraHop
+      })
+    } else {
+      // if the channel is disabled, we temporarily exclude it: this is necessary because the routing hint doesn't contain
+      // channel flags to indicate that it's disabled
+      data.c.assistedRoutes.flatMap(r => RouteCalculation.toChannelDescs(r, data.c.targetNodeId))
+        .find(_.shortChannelId == failure.update.shortChannelId)
+        .foreach(desc => router ! ExcludeChannel(desc))
```

Comment:
```suggestion
        .foreach(desc => router ! ExcludeChannel(desc)) // we want the exclusion to be router-wide so that sister payments in the case of MPP are aware the channel is faulty
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1441/commits/f3bb7e33cab0c5fe15343cc8e2d71aeb30122fd9